pkg/gofr/crud_handlers.go

```diff
@@ -61,7 +63,16 @@ type entity struct {
 
 // scanEntity extracts entity information for CRUD operations.
 func scanEntity(object interface{}) (*entity, error) {
-	entityType := reflect.TypeOf(object).Elem()
+	if object == nil {
+		return nil, errObjectIsNil
+	}
+
+	objType := reflect.TypeOf(object)
+	if objType.Kind() != reflect.Ptr {
+		return nil, errObjectNotPointer
```

Comment:
Can't we use `objType.Name()` here for a specialized error message in **errObjectNotPointer**

A error message like `failed to register routes for user struct, as it is passed by value should be pointer`

will be more informative from user's point of view

Reply:
Updated the error message and included the struct name as well